### ISSUE TITLE
ApacheHttpClient does not respect already encoded QueryMap(encoded = true)

### ISSUE THREAD

**Issue by antechrestos** [OpenFeign/feign#637]
Hi,

Using default client I am able to do something like this

```kotlin
interface MyService {
     @RequestLine("GET /something")
     fun getConnection(
            @QueryMap(encoded = true) filter: Map<String, String?>) : MyObject
}
```
Instantiating client like this

```kotlin
val objectMapper = ObjectMapper()
val builder = Feign.builder()
                .encoder(JacksonEncoder(objectMapper))
                .decoder(JacksonDecoder(objectMapper))
val client = builder.target(MyService::class.java, "http://somewhere.org")
```
Calling service like this

```kotlin
val result = client.getConnection(mapOf("foo" to "/some/value"))
```
Call the service with the following url: `http://somewhere.org/something?foo=/some/value `which is what I wanted.

I tried to use `ApacheHttpClient` 
```kotlin
val objectMapper = ObjectMapper()
val builder = Feign.builder()
                .client(ApacheHttpClient())
                ....
```
With this client, the url got is  `http://somewhere.org/something?foo=%2Fsome%2Fvalue `

How can I maintain the `@QueryMap(encoded = true)` ?

Thank you 😄

**Comment by kdavisk6** [comment]
I think I found the issue

https://github.com/OpenFeign/feign/blob/861bb8060c8d01c1fac47f5c1a6da28a0fa42479/httpclient/src/main/java/feign/httpclient/ApacheHttpClient.java#L105-L108

In the `ApacheHttpClient` it appears it is encoding all parameters all the time.  I'll do some more research.

**Comment by antechrestos** [comment]
Thank you very much @kdavisk6

**Comment by orwashere** [comment]
It looks like Apache Http Client is encoding the query parameters. Feign is using the RequestBuilder (Apache) to build the HttpUriRequest before making a request. `build` method within the RequestBuilder has the below code block handling the request parameters:
```

        if (this.parameters != null && !this.parameters.isEmpty()) {
            if (entityCopy != null || !"POST".equalsIgnoreCase(this.method) && !"PUT".equalsIgnoreCase(this.method)) {
                try {
                    uriNotNull = (new URIBuilder(uriNotNull)).setCharset(this.charset).addParameters(this.parameters).build();
                } catch (URISyntaxException var5) {
                    ;
                }
            } else {
                entityCopy = new UrlEncodedFormEntity(this.parameters, this.charset != null ? this.charset : HTTP.DEF_CONTENT_CHARSET);
            }
        }

        
```
If you check the 
`new URIBuilder(uriNotNull)).setCharset(this.charset).addParameters(this.parameters).build()`
addParameters mehod is handling the url encoding and there doesn't seem to be a way of changing this behaviour on the URIBuilder or RequestBuilder classes.
A stackoverflow answer mentions using BasicHttpRequest instead of HttpUriRequest but I need to dig deeper to see if this is a viable solution.

**Comment by kdavisk6** [comment]
This should be fixed via #778, but if this comes again, please feel free to re-open.

**Comment by hendryluk** [comment]
This is still happening, v10.2.3
